tests/test_document.py

```diff
@@ -0,0 +1,67 @@
+import os
+import tempfile
+
+import pytest
+
+from dangerzone.document import DocumentFilenameException, DocumentHolder
+
+from . import sample_doc, unreadable_pdf, unwriteable_pdf
+
+
+def test_input_sample_init(sample_doc):
+    DocumentHolder(sample_doc)
+
+
+def test_input_sample_after(sample_doc):
+    d = DocumentHolder()
+    d.input_filename = sample_doc
+
+
+def test_input_file_none():
+    """
+    Attempts to read a document's filename when no doc has been set
+    """
+    d = DocumentHolder()
+    with pytest.raises(DocumentFilenameException):
```

Comment:
When testing for exceptions, it's good to validate the exception message as well, especially for classes like `DocumentFilenameException` that can have multiple messages, to verify that the expected check was triggered. See `match=` in https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions13.